### PR TITLE
Release notes 1.2.0

### PR DESCRIPTION
**Pull request by hubertmis**

### CODE REVIEW

File: release_notes/nrf_802154_1.2.0.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# nRF IEEE 802.15.4 radio driver 1.2.0 release notes
+
+Release date: 2018-08-13
+
+This release contains new major feature: delayed transmission. The driver can schedule one frame to transmit it at requested time. It is used by higher-layer features like CSL, TSCH or ZigBee GP Proxy. The release also includes fixes to bugs found since the last release.
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: **a** new major feature , **a** requested time

File: release_notes/nrf_802154_1.2.0.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## New features:
+* New Radio Scheduler module resposible for radio preconditions management and delayed transmission scheduling. ([PR #106](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/106), [PR #107](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/107), [PR #108](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/108), [PR #111](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/111), [PR #112](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/112), [PR #113](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/113), [PR #114](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/114), [PR #115](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/115), [PR #120](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/120))
+* New module to manage delayed transmissions. ([PR #110](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/110), [PR #119](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/119))
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: for managing

File: release_notes/nrf_802154_1.2.0.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+* Renamed the Timer platform abstraction to the Low power (LP) timer. ([PR #122](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/122))
+* High precision (HP) timer platform abstraction used to provide exact timestamp of a received frame. ([PR #121](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/121), [PR #125](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/125))
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: - used -> is now used
- **the** exact timestamp

File: release_notes/nrf_802154_1.2.0.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+* Timer coordinator module resposible for coordination and synchronization between LP and HP timers. ([PR #123](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/123), [PR #126](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/126), [PR #127](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/127), [PR #142](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/142))
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: respo**n**sible

File: release_notes/nrf_802154_1.2.0.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+* Improved filtering of received frames destined to the PAN Coordinator node. ([PR #130](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/130))
+* Reduced number of compare channels used by simulator and SoftDevice RAALs. ([PR #124](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/124))
+* Reduced timeslot margin duration. ([PR #132](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/132))
+* New function to request transition to sleep state only if the radio is idle (i.e. idle listening). ([PR #143](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/143))
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: for requesting

File: release_notes/nrf_802154_1.2.0.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+## External dependencies changes:
+* Updated SoftDevice headers and documentation to version s140 v6.1.0. Note that the 802.15.4 radio driver is compatible with SoftDevice s140 v5.0.0, v6.0.0 and v6.1.0. ([PR #149](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/149))
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: v5.0.0, v6.0.0, and v6.1.0 (comma)

File: release_notes/nrf_802154_1.2.0.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+## Bug fixes:
+* Reduced power usage in the sleep state caused by TIMER peripherals. ([PR #105](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/105))
+* Passing correct NULL pointer value in the notification of transmitted frame. ([PR #109](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/109))
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: of **a** transmitted frame

File: release_notes/nrf_802154_1.2.0.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+* Fixed unaligned memory access in the extended address compare procedure. ([PR #104](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/104))
+* Corrected calculation of timeslot margin time if operating on LFClk with high PPM (i.e. RC oscilator). ([PR #147](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/147))
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: - if -> when
- oscil**l**ator

File: release_notes/nrf_802154_1.2.0.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+* Prevented double initialization of requested procedure on timeslot start. ([PR #116](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/116))
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: a requested procedure

File: release_notes/nrf_802154_1.2.0.md [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+* Fixed races related with the timer scheduler dependants. ([PR #117](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/117), [PR #118](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/118))
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: related to timer scheduler dependants.

File: release_notes/nrf_802154_1.2.0.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+* Prevented reentrancy of SWI notification functions - source of race conditions. ([PR #140](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/140))
+* Fixed double notification of received frame when timeslot ends during transmission of an Ack frame. ([PR #148](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/148))
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: - a received frame
- the timeslot

File: release_notes/nrf_802154_1.2.0.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+* Optimized time formats convertions. ([PR #150](https://github.com/NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver/pull/150))
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: time format conversions / conversions of time formats

File: release_notes/nrf_802154_1.2.0.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+
+## Limitations:
+* Number of supported peers for the Auto Pending Bit feature is limitted to 127.
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: limited

File: release_notes/nrf_802154_1.2.0.md [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+* (inherited) In multiprotocol configuration (SoftDevice RAAL), the application cannot use IRQ priority levels 2 and 3.
+
+## Known issues:
+* If `NRF_802154_USE_RAW_API` is disabled, a pointer to a frame passed by the transmission result notifications is invalid. Workaround: enable `NRF_802154_USE_RAW_API` if the code relies on the passed pointer.
+* Deinitialization during transmission procedure works incorrectly. Workaround: enter sleep state prior deinitialization.
```
Comment (b-gent) [inline release_notes/nrf_802154_1.2.0.md]: prior -> prior to / before

**Comment by b-gent** [comment]
please see my comments

**Comment by hubertmis** [comment]
Regression test

**Comment by hubertmis** [comment]
Regression test

**Pull request merged**